soh/src/code/z_parameter.c

```diff
@@ -1481,10 +1481,30 @@ void Inventory_SwapAgeEquipment(void) {
         }
 
         gSaveContext.adultEquips.equipment = gSaveContext.equips.equipment;
+        if (CVarGetInteger("gSwitchAge", 0) || CVarGetInteger("gSwitchTimeline", 0)) {
+            for (i = 0; i < ARRAY_COUNT(gSaveContext.equips.buttonItems); i++) {
+                gSaveContext.equips.buttonItems[i] = gSaveContext.childEquips.buttonItems[i];
+
+                if (i != 0) {
+                    gSaveContext.equips.cButtonSlots[i - 1] = gSaveContext.childEquips.cButtonSlots[i - 1];
+                }
+
+                if (((gSaveContext.equips.buttonItems[i] >= ITEM_BOTTLE) &&
+                     (gSaveContext.equips.buttonItems[i] <= ITEM_POE)) ||
+                    ((gSaveContext.equips.buttonItems[i] >= ITEM_WEIRD_EGG) &&
+                     (gSaveContext.equips.buttonItems[i] <= ITEM_CLAIM_CHECK))) {
+                    osSyncPrintf("Register_Item_Pt(%d)=%d\n", i, gSaveContext.equips.cButtonSlots[i - 1]);
+                    gSaveContext.equips.buttonItems[i] =
+                        gSaveContext.inventory.items[gSaveContext.equips.cButtonSlots[i - 1]];
+                }
+            }
 
-        if (gSaveContext.childEquips.buttonItems[0] != ITEM_NONE ||
-            CVarGetInteger("gSwitchAge", 0) ||
-            CVarGetInteger("gSwitchTimeline", 0)) {
+            gSaveContext.equips.equipment = gSaveContext.childEquips.equipment;
+            gSaveContext.equips.equipment &= 0xFFF0;
+            if (!((1 << 0 & gSaveContext.inventory.equipment) == 0 || gSaveContext.infTable[29] & 1)) {
+                gSaveContext.equips.equipment |= 0x0001;
+            }
```

Comment:
Can we add a comment here explaining what this is doing? I'm having trouble parsing what exactly is happening here, beyond a general sense of enabling/disabling equipment according to certain conditions.

Reply:
Yea will add comments, this just avoids the issue of having kokiri sword equipped in the inventory subscreen even when link is swordless.

Reply:
OK I think I see, so it unequips Kokiri sword with `gSaveContext.equips.equipment &= 0xFFF0;` and re-equips it if applicable with `gSaveContext.equips.equipment |= 0x0001;`?

Reply:
Yeah, I think I'll change it to two NOT conditions and AND them so its more clear of the two cases where the sword is not equipped